homeassistant/components/samsungtv/media_player.py

```diff
@@ -129,12 +128,13 @@ def get_remote(self):
                 self._remote = SamsungRemote(self._config.copy())
             # This is only happening when the auth was switched to DENY
             # A removed auth will lead to socket timeout because waiting for auth popup is just an open socket
-            except AccessDenied:
+            except samsung_exceptions.AccessDenied:
                 self.hass.async_create_task(
                     self.hass.config_entries.flow.async_init(
                         DOMAIN, context={"source": "reauth"}, data=self._config_entry
                     )
                 )
+                raise samsung_exceptions.AccessDenied()
```

Comment:
Just `raise`. That will raise the caught exception again without loosing context.